.github/workflows/lint-server.yaml

```diff
@@ -0,0 +1,44 @@
+---
+# This workflow will run Spotless with Java configuration to format the Java codebase
+name: Lint Server
+# yamllint disable-line rule:truthy
+on:
+  workflow_call:
+  push:
+    paths:
+      - 'packages/server/**'
+    branches: [ 'main' ]
+  pull_request:
+    paths:
+      - 'packages/server/**'
+    branches: [ 'main' ]
+
+jobs:
+  lintServer:
+    name: Lint Server
+    runs-on: ubuntu-22.04
+
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the
+      # added or changed files to the repository.
+      contents: write
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Setup Java v17
+        uses: actions/setup-java@v4
+        with:
+          distribution: "temurin"
+          java-version: 17
+          cache: gradle
+
+      - name: Run Spotless
+        run: |
+          ./gradlew spotlessApply
+
+      - name: Commit all changed files
+        uses: stefanzweifel/git-auto-commit-action@v5
```

Comment:
please switch to use a `sha` for the version

Reply:
and you sure you want to auto-commit these?

Reply:
Done. And yes I want to auto-commit those, it's always annoying to come back to re-run something that already run on the github action. Also it's easy to remove a commit from the branch if necessary.